docs/howto/static-files/index.txt

```diff
@@ -108,6 +108,16 @@ do this by adding the following snippet to your urls.py::
     folder; it doesn't perform static files discovery like
     :mod:`django.contrib.staticfiles`.
 
+.. note:: 
+     
+    During development, we know that serving files could be done automatically;
+    it doesn't go through the middleware chain when we set the DEBUG is True. 
+    What it does is that it enables a view function that serves our static
+    files automatically. But it's not the whole process. Runserver usually
+    uses WSGI_APPLICATION, which in turn depends on middleware functions. 
+    But when Debug is true, runserver overrides the core functionality per
+    se. And it adds an automatic serving of static files; the file serving 
+    doesn't go through MIDDLEWARE.  
```

Comment:
I don't think going into the exact implementation is called for… 🤔

I had something in mind more like... 

```
    Static files are served via a wrapper at the WSGI application layer, and
    static files requests do not pass through the normal middleware chain. 
```

I think this could added to the existing `note` block immediately above, perhaps beginning, `Finally, …` 🤔 

(Also the block should have a single blank line following it for formatting.)

Reply:
I'll make sure to add the following lines. Mr Carlton, I know you are busy at the moment because of the Django 4.0 release. But when I was working with this ticket. I saw a similar ticket #27325 [https://code.djangoproject.com/ticket/27325](url). I'll summarize this for you in two lines. **Offer a solution for static file serving suitable for production.** But people tried this five years ago and they nearly came with a solution but didn't really apply it. You know what is best for Django's growth, do you think this will add any value to Django.

Reply:
Hi @valleyofblackpanther If ticket-27325 were to be resolved, we'd remove the added text here as part of that. 

But, as you note, it's been a few years already. It may be a few more yet, so this note is intended to help folks in the meantime. (I'd guess that'd be @kezabelle's take opening the issue…)

Reply:
Hi @carltongibson. So you are meaning to tell that @kezabelle is responsible to open the issue again.

Reply:
I'm pleased to see someone picking up this documentation ticket, so thanks :)

Carlton's correct: for my purposes (i.e: [ticket 33048](https://code.djangoproject.com/ticket/33048) being worked on here) documenting the special-casing is enough (actually for my purposes I just monkeypatch'd some stuff, and now the problem no longer exists for me ;)).

The other ticket, [27325](https://code.djangoproject.com/ticket/27325) is not my call, but would one day be good to finally see land (personally if it were _me_, a starting point would be to convert `StaticFilesHandlerMixin` into a middleware, which isn't insurmountably difficult, and promote static-files serving to be a MW, and then later co-ordinate introducing WhiteNoise to replace _that_, YMMV).

Reply:
Okay, thanks for the instruction @kezabelle. Will look into that.